### ISSUE TITLE
disable parsing version.txt globally for a repo?

### ISSUE THREAD

**Issue by japj** [dotnet/Nerdbank.GitVersioning#956]
Hello,

We have an existing codebase where version.txt files exist (that are not compatible with nerdbank.gitversioning, which causes it to fail).
We will use version.json files to configure nerdbank.gitversioning.

Is it possible to globally disable the use of version.txt by nerdbank.gitversioning?

**Comment by AArnott** [comment]
Not at present, but your case makes sense to add a way to disable them.

**Comment by AArnott** [comment]
I wonder how they would be disabled though. A version.json setting? We look for version.txt and version.json from the current or project directory up toward the repo root, so if we hit a version.txt first, before we read the json that disables it, that becomes a problem. 
An environment variable would be simple to implement, but is an onerous requirement to put on all the developers of your repo. 
If your only use is through msbuild via Nerdbank.GitVersioning, an msbuild property could be made to do it, which you set in Directory.Build.props at the root of your repo. That would work, but it wouldn't fix the `nbgv` CLI tool scenario, if you have one. If you need it, we could add a switch to that tool to ignore version.txt files. But you'd have to add that every time you use it, unless we come up with another clever idea.

**Comment by japj** [comment]
I believe we only use it from MSBuild, so a property would be sufficient for now

**Comment by AArnott** [comment]
I haven't been able to reserve time for this yet. I'm not sure when I will. 
Would you be interested in submitting the PR yourself? 

Alternatively...
Sponsorship may be an effective way to lead a contributor of this repo to prioritize the feature. As a free and open source library, all or the majority of its development is done without any monetary benefit to its authors, and all of it is made available free of charge. But sponsorship is very much appreciated and, with agreement from a particular contributor and/or the owning team, can be a great way to support the project and help get your favorite features or bugs addressed in an upcoming version.
If sponsorship to drive completion of a feature or bug fix is an interesting option to you, please let us know. You can reach out to me privately on [keybase.io](https://keybase.io/aarnott) or [gitter.im](https://gitter.im/aarnott) as well.

**Comment by japj** [comment]
Hi, 

for the moment we managed to workaround it by not having a version.txt in the same folder (or parent structure) as a C# project file without breaking existing product logic. In essence although we identified multiple version.txt files in the source tree, only a limited few were causing "havoc" in the build and we could rename those to .md.

That seems to be good enough for us at the moment, as focus in this specific case is on migrating an existing code base to git (which is interesting enough as it is). 

If we run into another issue during this migration that needs to be fixed inside Nerdbank.GitVersioning I will definitely create a PR myself.
